lib/evmone/eof.cpp

```diff
@@ -220,14 +226,14 @@ std::variant<std::vector<EOFCodeType>, EOFValidationError> validate_types(
     assert(!container.empty());  // guaranteed by EOF headers validation
 
     std::vector<EOFCodeType> types;
+    types.reserve(type_section_size / 4);
 
     // guaranteed by EOF headers validation
     assert(header_size + type_section_size < container.size());
 
-    for (auto offset = header_size; offset < header_size + type_section_size; offset += 4)
+    for (size_t i = 0; i < type_section_size / 4; ++i)
     {
-        types.emplace_back(
-            container[offset], container[offset + 1], read_uint16_be(&container[offset + 2]));
+        types.emplace_back(get_type(container, header_size, i));
```

Comment:
However, this is not what was intended. We want to drop `std::vector<EOFCodeType> types`. And we want to replace all uses of  `header.types[fid]` with `header.get_type(container, index)`.

Hint: you will have to pass the container to the `EOF1Header::get_type()` because the header object don't have the reference to the container. We may change this later, but that's separate topic.

Reply:
Originally I had the method in the header class passing in the container, but I re read the issue and tried to mimic the suggested function signature.

I'll make the change

Reply:
Sorry for the confusion then :)

Reply:
@chfast Fixed it